### PR TITLE
Complain about missing Yasm at configure time at of build time

### PR DESCRIPTION
**Pull request by rschu1ze**
I made a local build with ENABLE_LIBRARIES=1 and got funny errors at build time about yasm was not found. Running this check now earlier during CMake configure.

Follow-up to #44949

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by rschu1ze** [comment]
@kssenii Since you reviewed #44949, perhaps you could check this one too?

File: contrib/libhdfs3-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -105,6 +105,9 @@ set(SRCS
 
 if (ARCH_AMD64)
     find_program(YASM_PATH NAMES yasm)
+    if (NOT YASM_PATH)
+        message(FATAL_ERROR "Please install the Yasm assembler")
```
Comment (kssenii) [inline contrib/libhdfs3-cmake/CMakeLists.txt:109]: maybe disable building hdfs in this case after logging the message like "Not using hdfs because ..."?

Comment (rschu1ze) [reply contrib/libhdfs3-cmake/CMakeLists.txt:109]: I actually don't like that too much. Nobody reads the log messages of all >50 contribs and there is too much potential for surprises. I also checked for other contribs which do `find_program` and could only find "krb5" (which needs awk) and if the dependency is not found CMake configuration simply fails.

Comment (kssenii) [reply contrib/libhdfs3-cmake/CMakeLists.txt:109]: Ok

**Review by kssenii — APPROVED** [review]

**Pull request merged**